src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/goal/service/GoalService.kt

```diff
@@ -42,12 +42,14 @@ class GoalService(
   }
 
   /**
-   * Updates a [Goal] identified by its `prisonNumber` and `goalReference`.
-   * Throws [GoalNotFoundException] if the [Goal] cannot be found.
+   * Updates a [Goal], identified by its `prisonNumber` and `goalReference`, from the specified `updatedGoal`.
+   * Throws [GoalNotFoundException] if the [Goal] to be updated cannot be found.
    */
-  fun updateGoal(prisonNumber: String, goalReference: UUID): Goal {
+  fun updateGoal(prisonNumber: String, goalReference: UUID, updatedGoal: Goal): Goal {
```

Comment:
The service method passes the prison number, goal reference, and proposed updated goal into the Persistence Adapter method (the method above ☝️ that is marked `@Transactional`)